File: apps/desktop/src/lib/api/projects.ts
/**
 * Project management API
 * Communicates with Rust backend via Tauri IPC
 */

import { invoke } from '@tauri-apps/api/core';
import type { Project, CreateProjectInput } from '@aycd/core';

/**
 * Creates a new AYCD project
 */
export async function createProject(input: CreateProjectInput): Promise<Project> {
  return await invoke<Project>('create_project', {
    name: input.name,
    customPath: input.path,
  });
}

/**
 * Opens an existing project by path
 */
export async function openProject(projectPath: string): Promise<Project> {
  return await invoke<Project>('open_project', { projectPath });
}

/**
 * Lists all projects in the default directory
 */
export async function listProjects(): Promise<Project[]> {
  return await invoke<Project[]>('list_projects');
}

/**
 * Gets the default projects root directory path
 */
export async function getProjectsRoot(): Promise<string> {
  return await invoke<string>('get_projects_root');
}

/**
 * Updates project metadata
 */
export async function updateProject(project: Project): Promise<void> {
  await invoke('update_project', { project });
}

File: apps/desktop/src-tauri/src/commands/projects.rs
use crate::models::Project;
use crate::services::project_service;
use std::path::PathBuf;

/// Creates a new AYCD project
#[tauri::command]
pub async fn create_project(name: String, custom_path: Option<String>) -> Result<Project, String> {
    let path = custom_path.map(PathBuf::from);

    project_service::create_project(&name, path)
        .map_err(|e| format!("Failed to create project: {}", e))
}

/// Opens an existing project
#[tauri::command]
pub async fn open_project(project_path: String) -> Result<Project, String> {
    let path = PathBuf::from(project_path);

    project_service::open_project(&path)
        .map_err(|e| format!("Failed to open project: {}", e))
}

/// Lists all projects in the default directory
#[tauri::command]
pub async fn list_projects() -> Result<Vec<Project>, String> {
    project_service::list_projects()
        .map_err(|e| format!("Failed to list projects: {}", e))
}

/// Gets the default projects root directory path
#[tauri::command]
pub async fn get_projects_root() -> Result<String, String> {
    project_service::get_projects_root()
        .map(|p| p.to_string_lossy().to_string())
        .map_err(|e| format!("Failed to get projects root: {}", e))
}

/// Updates project metadata
#[tauri::command]
pub async fn update_project(project: Project) -> Result<(), String> {
    project_service::update_project(&project)
        .map_err(|e| format!("Failed to update project: {}", e))
}
